src/util/downloadUtils.ts

```diff
@@ -98,3 +99,46 @@ export async function downloadFromURL(
       });
   });
 }
+
+/**
+ * Check if a given url returns an expected status code.
+ */
+export async function hasStatusCode(
+  url: URL,
+  statusCode: number
+): Promise<boolean> {
+  return new Promise(resolve => {
+    const transporter = url.protocol === 'http:' ? http : https;
+
+    const request = transporter
+      .get(url, { timeout: SERVER_STATUS_CHECK_INTERVAL }, res => {
```

Comment:
If we're just checking to see if we can access this, we should be using the [HEAD](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/HEAD) or OPTIONS method instead. Both don't require re-downloading the JSAPI (which is also not cached, which may be another reason this polling is a bad idea).

Reply:
I went with OPTIONS since the response seemed slightly smaller than HEAD